rules/xcodeproj.bzl

```diff
@@ -63,6 +63,29 @@ def _xcodeproj_aspect_collect_hmap_paths(deps, target, ctx):
                 hmap_paths.append(relative_path)
     return hmap_paths
 
+def _targeted_device_family(ctx):
+    """ The targeted device family of the rule
+
+    Args:
+        ctx: same as what is passed into aspect impl
+    Returns:
+        the targeted device family in the format recognized by xcode
+    """
+    targeted_device_family = None
+    for device_family in getattr(ctx.rule.attr, "families", []):
```

Comment:
I think this loop could be simplified as

```
mapping = {"iphone": "1", "ipad":"2"}
",".joined([mapping[family] if family in mapping else fail() for family in getattr(ctx.rule.attr, "families", [])])
```

or something more funtional like that